core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -362,10 +381,39 @@ object Chunk {
       Bytes(values, offset, n) -> Bytes(values, offset + n, length - n)
     override def map[O2](f: Byte => O2): Chunk[O2] = seq(values.map(f))
     override def toArray[O2 >: Byte: ClassTag]: Array[O2] = values.slice(offset, offset + length).asInstanceOf[Array[O2]]
-    def toByteBuffer: ByteBuffer = ByteBuffer.wrap(values, offset, length)
   }
   object Bytes { def apply(values: Array[Byte]): Bytes = Bytes(values, 0, values.length) }
 
+  /** Creates a chunk backed by an byte buffer, bounded by the current position and limit */
+  def byteBuffer(buf: JByteBuffer): Chunk[Byte] = ByteBuffer(buf)
+
+  final case class ByteBuffer private (buf: JByteBuffer, offset: Int, size: Int) extends Chunk[Byte] {
+    def apply(i: Int): Byte = buf.get(i + offset)
+    protected def splitAtChunk_(n: Int): (Chunk[Byte], Chunk[Byte]) = {
+      val first = buf.asReadOnlyBuffer
+      first.limit(n + offset)
+      val second = buf.asReadOnlyBuffer
+      second.position(n + offset)
+      (ByteBuffer(first), ByteBuffer(second))
+    }
+    override def map[O2](f: Byte => O2): Chunk[O2] = {
```

Comment:
I haven't benchmarked this against other plausible implementations.  I think this will be faster than bulk-getting the array and mapping that, but I haven't proven it.